### ISSUE TITLE
Slow searching through zip files

### ISSUE THREAD

**Issue by raelb** [dnGrep/dnGrep#656]
Hi,
Perhaps this is something that can be improved.

Search in archives (zip files) with dnGrep took around 50s.
Some other tools took only a couple of seconds (ugrep)

Can't upload file, but here is dropbox link:
https://www.dropbox.com/s/lcjuzdgm5zropk6/Firefox%20extensions.zip?dl=0

Just extract to a folder. I was searching for text `copy all`

![image](https://user-images.githubusercontent.com/13108198/160293523-718f6839-d67c-4bdf-a098-6cb832239023.png)

![image](https://user-images.githubusercontent.com/13108198/160293538-9c573130-47ce-44a1-b601-4fe5b417a253.png)

**Comment by doug24** [comment]
Yes, it looks like it has gone wrong here.  I think I know what it is, but it's going to take some rework to fix it.

**Comment by doug24** [comment]
Making progress on this:

![capture-000072](https://user-images.githubusercontent.com/17227248/162586286-73bed4e8-fb80-48b3-b969-056d0996023e.png)

**Comment by raelb** [comment]
Looks good to me ☺

**Comment by doug24** [comment]
This is released now in v3.0.64
